modules/openapi-generator/src/main/java/org/openapitools/codegen/OpenAPINormalizer.java

```diff
@@ -0,0 +1,384 @@
+            ComposedSchema m = (ComposedSchema) schema;
+            if (m.getAllOf() != null && !m.getAllOf().isEmpty()) {
+                normalizeAllOf(m, visitedSchemas);
+            }
+
+            if (m.getOneOf() != null && !m.getOneOf().isEmpty()) {
+                normalizeOneOf(m, visitedSchemas);
+            }
+
+            if (m.getAnyOf() != null && !m.getAnyOf().isEmpty()) {
+                normalizeAnyOf(m, visitedSchemas);
+            }
+
+            if (m.getProperties() != null && !m.getProperties().isEmpty()) {
+                normalizeProperties(m.getProperties(), visitedSchemas);
+            }
+
+            if (m.getAdditionalProperties() != null) {
+                // normalizeAdditionalProperties(m);
+            }
+        } else if (schema.getNot() != null) {// not schema
+            normalizeSchema(schema.getNot(), visitedSchemas);
+        } else if (schema.getProperties() != null && !schema.getProperties().isEmpty()) {
+            normalizeProperties(schema.getProperties(), visitedSchemas);
+        } else if (schema instanceof Schema) {
+            normalizeNonComposedSchema(schema, visitedSchemas);
+        } else {
+            throw new RuntimeException("Unknown schema type found in normalizer: " + schema);
+        }
+    }
+
+    private void normalizeNonComposedSchema(Schema schema, Set<Schema> visitedSchemas) {
+        // normalize non-composed schema (e.g. schema with only properties)
+    }
+
+    private void normalizeProperties(Map<String, Schema> properties, Set<Schema> visitedSchemas) {
+        if (properties == null) {
+            return;
+        }
+        for (Map.Entry<String, Schema> propertiesEntry : properties.entrySet()) {
+            Schema property = propertiesEntry.getValue();
+            normalizeSchema(property, visitedSchemas);
+        }
+    }
+
+    private void normalizeAllOf(Schema schema, Set<Schema> visitedSchemas) {
+        for (Object item : schema.getAllOf()) {
+            if (!(item instanceof Schema)) {
+                throw new RuntimeException("Error! allOf schema is not of the type Schema: " + item);
+            }
+            // normalize allOf sub schemas one by one
+            normalizeSchema((Schema) item, visitedSchemas);
+        }
+        // process rules here
+        processUseAllOfRefAsParent(schema);
+    }
+
+    private void normalizeOneOf(Schema schema, Set<Schema> visitedSchemas) {
+        for (Object item : schema.getAllOf()) {
+            if (!(item instanceof Schema)) {
+                throw new RuntimeException("Error! allOf schema is not of the type Schema: " + item);
+            }
+            // normalize oenOf sub schemas one by one
+            normalizeSchema((Schema) item, visitedSchemas);
+        }
+        // process rules here
+    }
+
+    private void normalizeAnyOf(Schema schema, Set<Schema> visitedSchemas) {
+        for (Object item : schema.getAllOf()) {
```

Comment:
Shouldn't this be `getAnyOf()`?

Reply:
yes, I noticed that too. I'm working on another rule. Will fix this.

Reply:
should be fixed by https://github.com/OpenAPITools/openapi-generator/pull/14463